addons/web/static/tests/views/form_tests.js

```diff
@@ -5155,5 +5121,33 @@ QUnit.module('Views', {
 
         form.destroy();
     });
+
+    QUnit.test('rainbowman attributes correctly passed on button click', function (assert) {
+        assert.expect(1);
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<header>' +
+                        '<button name="action_won" string="Won" type="object" rainbow="{\'message\': \'Congrats!\'}"/>' +
+                    '</header>' +
+                '</form>',
+            intercepts: {
+                execute_action: function (event) {
+                    var rainbow_data = pyeval.py_eval(event.data.action_data.rainbow);
```

Comment:
here, you should just test that the rainbow data is properly passed. the test should not instantiate a rainbowman itself